### PR TITLE
Try to prevent a deadlock when writing options

### PR DESCRIPTION
**Pull request by tsteur**
refs https://github.com/matomo-org/matomo/issues/15545#issuecomment-594927218

Invalidator tests passed for me locally. Not sure I missed something maybe?

### CODE REVIEW

File: core/Archive/ArchiveInvalidator.php [hunk 1]
```diff
@@ -78,11 +78,12 @@ public function getAllRememberToInvalidateArchivedReportsLater()
         // we do not really have to get the value first. we could simply always try to call set() and it would update or
         // insert the record if needed but we do not want to lock the table (especially since there are still some
         // MyISAM installations)
-        $values = Option::getLike($this->rememberArchivedReportIdStart . '%');
+        $values = Option::getLike('%' . $this->rememberArchivedReportIdStart . '%');
```
Comment (diosmosis) [inline core/Archive/ArchiveInvalidator.php]: Sometimes I see `'%_' .` and other times `'%' .`, could that be an issue?

Comment (tsteur) [reply core/Archive/ArchiveInvalidator.php]: @diosmosis not really. I just copied behaviour what was done on the right part, to have the same on the left. It'll not make a difference though

**Comment by diosmosis** [comment]
I don't see a change to the `Option::getLike()` in `deleteOptionLike()`, should there be one?

**Comment by tsteur** [comment]
@diosmosis yes there should! Will add it now. I used to have it there but must have been undone by some other reverts

**Comment by tsteur** [comment]
@diosmosis actually it works because it is already partially sent with the `%` when the method is called. I will make that more clear. It's confusing right now

**Comment by diosmosis** [comment]
looks good to merge if tests pass 👍

**Pull request merged**